### ISSUE TITLE
Generalisation to regression 

### ISSUE THREAD

**Issue by James-Large** [sktime/sktime-dl#18]
A reasonable extension that fits back into the base sktime would be to generalise the networks to be usable for regression out the box as well, in addition to any reduction techniques via base sktime 

Fundamentally, this should be a matter of changing the output layer and loss function, maybe a couple more finer details also

Likely, this should involve a refactor separating the 'actual' network definitions (e.g. build model minus the final layer, and any bespoke fitting/training code) into their own private sections, with wrappers to add on classification/regression/(likely other tasks in future) capabilities 

e.g. file structure

- networks
  - _ResNetNetwork: defines build model, minus output layer and model compilation
  - ...
- classifiers 
  - ResNetClassifier: builds a _ResNetNetwork instance, adds output/compiles, ready for fit/predict
  - ...
- regressors 
  - ResNetRegressor: "   "
  - ...

**Comment by James-Large** [comment]
Update to this, most networks have had this generalisation process applied in #22. 

Generalised: 

* Time convolutional neural network (CNN)
* Encoder (Encoder)
* Fully convolutional neural network (FCNN)
* Multi layer perceptron (MLP)
* Residual network (resnet)
* Time Le-Net (tlenet)
* InceptionTime (inception)

Some remain due to more bespoke training procedures, but which are in principle certainly still possible: 

* Multi channel deep convolutional neural network (MCDCNN)
* Multi-scale convolutional neural network (MCNN)
* Time warping invariant echo state network (twiesn)

**Comment by mloning** [comment]
We should also generalise the meta-estimators, tuning and ensembling should work with both regressors and classifiers.

**Comment by Withington** [comment]
I have MCDCNN regressor in my fork. I'll wait until check_fitted #39 is merged to dev before creating the pull request.
